### PR TITLE
sql: reconcile expression simplification with type checking

### PR DESCRIPTION
**Pull request by tamird**
Seeing some strange failures in select_test. @petermattis any ideas?

### CODE REVIEW

**Comment by tamird** [comment]
Fixed.

File: sql/analyze_test.go [hunk 1]
```diff
@@ -31,10 +31,14 @@ func testTableDesc() *TableDescriptor {
 		ID:   1000,
 		Columns: []ColumnDescriptor{
 			{Name: "a", Type: ColumnType{Kind: ColumnType_INT}},
-			{Name: "b", Type: ColumnType{Kind: ColumnType_INT}},
-			{Name: "c", Type: ColumnType{Kind: ColumnType_INT}},
-			{Name: "d", Type: ColumnType{Kind: ColumnType_INT}},
-			{Name: "e", Type: ColumnType{Kind: ColumnType_INT}},
+			{Name: "b", Type: ColumnType{Kind: ColumnType_BOOL}},
```
Comment (petermattis) [inline sql/analyze_test.go]: I think some of the tests in `select_test.go` relied on these columns being INTs.

Comment (petermattis) [inline sql/analyze_test.go]: Ah, I see you've changed those tests to use column `i`. You could have avoided touching those tests by leaving column `b` with type INT. Oh well, doesn't matter now.

**Comment by petermattis** [comment]
LGTM

File: sql/analyze_test.go (hunk 1 above)
Comment (tamird) [inline sql/analyze_test.go]: backed that out, the diff is smaller this way.

**Pull request merged**